src/test/java/org/mule/test/extension/file/common/AbstractPostActionGroupTestCase.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.extension.file.common;
+
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.mule.extension.file.common.api.AbstractFileSystem;
+import org.mule.extension.file.common.api.FileAttributes;
+import org.mule.extension.file.common.api.FileConnectorConfig;
+import org.mule.extension.file.common.api.command.*;
+import org.mule.extension.file.common.api.exceptions.FileAlreadyExistsException;
+import org.mule.extension.file.common.api.lock.PathLock;
+import org.mule.extension.file.common.api.source.AbstractPostActionGroup;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.nio.file.Path;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractPostActionGroupTestCase {
```

Comment:
I would assert on each case where the apply doesn't fail, what action is done. Was the file deleted? was it moved? was it renamed?